src/modules/sensors/sensors.cpp

```diff
@@ -1461,11 +1461,9 @@ Sensors::parameter_update_poll(bool forced)
 
 		close(fd);
 
-		fd = open(AIRSPEED_DEVICE_PATH, 0);
-
 		/* this sensor is optional, abort without error */
-
-		if (fd > 0) {
+		fd = open(AIRSPEED_DEVICE_PATH, 0);
+		if (fd != -1) {
```

Comment:
Although this might be correct against POSIX specs, NuttX sometimes returns -ERRNO to some calls. The check should therefore be < 0 or >= 0, not against a particular value.